### PR TITLE
[REF] l10n_ec: Refactor _l10n_ec_get_identification_type

### PR DESCRIPTION
**Pull request by antoine162**
`_l10n_ec_get_identification_type` will get called quite often in a loop in the ATS report, so it's important that it gets orm-cached. This reimplementation removes calls to `ref()` and replaces them with calls to `_xmlid_to_res_model_res_id()` which is entirely orm-cached.

### CODE REVIEW

File: addons/l10n_ec/models/res_partner.py [hunk 1]
```diff
@@ -101,15 +101,26 @@ def _l10n_ec_get_identification_type(self):
...
+            for xmlid in ec_id_types_by_xmlid
+        }
+
+        if (
+            (xmlid := xmlid_by_res_id.get(self.l10n_latam_identification_type_id.id))
+            and (ec_id_type := ec_id_types_by_xmlid.get(xmlid))
+        ):
+            return ec_id_type
+
+        if self.l10n_latam_identification_type_id.country_id != self.env.ref('base.ec'):
```
Comment (jco-odoo) [inline addons/l10n_ec/models/res_partner.py:125]:
```suggestion
        if self.l10n_latam_identification_type_id.country_id.code != 'EC':
```

Comment (jco-odoo) [reply addons/l10n_ec/models/res_partner.py:125]: walrus in 16?

Comment (antoine162) [reply addons/l10n_ec/models/res_partner.py]: Thanks :smile:

**Comment by antoine162** [comment]
@jco-odoo btw, it seems the walrus operator was introduced in python 3.8 :smile:

**Comment by jco-odoo** [comment]
@robodoo r+

**Pull request closed without merging**